bokehjs/src/lib/models/tools/actions/save_tool.ts

```diff
@@ -24,7 +24,15 @@ export class SaveToolView extends ActionToolView {
   doit(action: "save" | "copy" = "save"): void {
     switch (action) {
       case "save":
-        this.save("bokeh_plot")
+        if (this.model.filename == "bokeh_plot") {
```

Comment:
I don't love conditioning on a concrete default value. Perhaps it would be better to have a `Nullable(String...)` property with a null (None) default, then this could condition on the value being null.

Reply:
I think this is now addressed in 36ed47e but feel free to comment.